stubs/dom.stub

```diff
@@ -68,6 +76,44 @@ class DOMNodeList implements Traversable
 
 }
 
+class DOMXPath
+{
+
+	/** @var DOMDocument */
+	public $document;
+
+    /**
+     * @param string $expression
+     * @param DOMNode|null $contextNode
+     * @param boolean $registerNodeNS
+     * @return boolean
+     */
+    public function evaluate($expression, $contextNode, $registerNodeNS) {}
+
+    /**
+     * @template TNode as DOMNode
```

Comment:
I'm unsure if this is the best way to do this? I had originally just `@return DOMNodeList|false` below, which seemed to work quite well in my testing, but that caused the PHPStan check to fail, wanting a specific type, not the generic. So I did this, but it seems redundant to the template in `DOMNodeList`.

What I think this should say is that it can return any valid specialization of `DOMNodeList`, which in turn is any subclass of DOMNode, and that there is no way to infer which one exactly, but if no further info is provided, it should be treated as `DOMNodeList<DOMNode>` - so if there is a better way to state that, please let me know.

Reply:
Hmm when I run `./bin/phpstan analyze --level=max ./stubs/dom.stub` or just `make phpstan` it doesn't complain, but the test run automatically in github seems to disapprove of this.

Reply:
Oh I see it's `make phpstan-validate-stub-files` that complains. At least now I can check it locally.

`Template type TNode of method DOMXPath::query() is not referenced in a parameter.`

So it doesn't want the return type to be a generic, and doesn't want a template parameter that it can't infer from an input, I understand, but how do I tell it that the return can be any valid specialization of DOMNodeList? Should it just use `@return DOMNodeList<DOMNode>` I guess?